### PR TITLE
added documentation about the format of certificateKey

### PR DESCRIPTION
**Pull request by SaumyaBhushan**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
add more documentation about the format of certificateKey
#### Which issue(s) this PR fixes:

xref https://github.com/kubernetes/kubeadm/issues/2928

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by SaumyaBhushan** [comment]
Hi @neolit123 ! Updated all the changes here. Please review .

**Comment by neolit123** [comment]
please keep the commits squashed to one.

**Comment by neolit123** [comment]
in the PR description change:
Fixes https://github.com/kubernetes/kubeadm/issues/2928

to:
xref https://github.com/kubernetes/kubeadm/issues/2928

or just:
https://github.com/kubernetes/kubeadm/issues/2928

otherwise it will close the ticket.

File: cmd/kubeadm/app/apis/kubeadm/v1beta3/types.go [hunk 1]
```diff
@@ -53,7 +53,8 @@ type InitConfiguration struct {
 	LocalAPIEndpoint APIEndpoint `json:"localAPIEndpoint,omitempty"`
 
 	// CertificateKey sets the key with which certificates and keys are encrypted prior to being uploaded in
-	// a secret in the cluster during the uploadcerts init phase.
+	// a secret in the cluster during the uploadcerts init phase.
```
Comment (neolit123) [inline cmd/kubeadm/app/apis/kubeadm/v1beta3/types.go:56]: why is this line changed?

File: cmd/kubeadm/app/cmd/join.go [hunk 2]
```diff
@@ -251,7 +251,7 @@ func addJoinConfigFlags(flagSet *flag.FlagSet, cfg *kubeadmapiv1.JoinConfigurati
 	)
 	flagSet.StringVar(
 		&cfg.ControlPlane.CertificateKey, options.CertificateKey, cfg.ControlPlane.CertificateKey,
-		"Use this key to decrypt the certificate secrets uploaded by init.",
+		"Use this key to decrypt the certificate secrets uploaded by init.The certificate key is a hex encoded string that is an AES key of size 32 bytes.",
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/join.go:254]:
```suggestion
		"Use this key to decrypt the certificate secrets uploaded by init. The certificate key is a hex encoded string that is an AES key of size 32 bytes.",
```

File: cmd/kubeadm/app/apis/kubeadm/v1beta3/types.go (hunk 1 above)
Comment (SaumyaBhushan) [reply cmd/kubeadm/app/apis/kubeadm/v1beta3/types.go:56]: Not sure about this one .

**Comment by SaumyaBhushan** [comment]
Done @neolit123 . please review .

File: cmd/kubeadm/app/apis/kubeadm/v1beta3/types.go (hunk 1 above)
Comment (neolit123) [reply cmd/kubeadm/app/apis/kubeadm/v1beta3/types.go:56]: they are the same, byte-by-byte.
strange.

**Comment by neolit123** [comment]
/ok-to-test
/triage accepted
/approve
/lgtm

thanks!

**Comment by neolit123** [comment]
failing test:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/120546/pull-kubernetes-verify/1700803651615854592

please run ./hack/update-gofmt.sh and commit the changes (keep it single commit). make sure your setup is for Go 1.20x

**Comment by SaumyaBhushan** [comment]
Hi @neolit123 ! Now tests are passing successfully

**Review by neolit123 — COMMENTED** [review]
/lgtm

**Comment by sftim** [comment]
This change looks like it'd be visible to end users. I recommend adding a changelog entry.
